addons/udes_stock_routing/models/stock_move.py

```diff
@@ -337,55 +336,40 @@ def _do_unreserve(self):
                     continue
                 split_pick_type = rule.picking_type_id
                 original_picking_type = rule.u_run_on_assign_applicable_to
+                if split_pick_type == original_picking_type:
+                    _logger.debug(
+                        "Skipping because split and original picking types are the same: %r",
+                        split_pick_type | original_picking_type,
+                    )
+                    continue
                 split_pick_moves = moves.filtered(lambda p: p.picking_type_id == split_pick_type)
```

Comment:
I'd move this line before the if and consider doing: split_pick_moves._action_confirm(merge=True) inside the if

Reply:
Just thinking that the `split_pick_moves._action_confirm(merge=True)` would be better to put in a method so the behaviour can be overridden if necessary.